tools/golangci-lint.sh

```diff
@@ -26,31 +21,7 @@ ensure_binary() {
     echo "info: Downloading ${NAME} ${RELEASE} to build environment"
 
     mkdir -p "${BUILD_BIN}"
-
-    case "${OSTYPE}" in
-      "darwin"*) os_type="darwin"; sum="${OSX_RELEASE_256}" ;;
-      "linux"*) os_type="linux"; sum="${LINUX_RELEASE_256}" ;;
-      *) echo "error: Unsupported OS '${OSTYPE}' for shellcheck install, please install manually" && exit 1 ;;
-    esac
-
-    release_archive="/tmp/${NAME}-${RELEASE}.tar.gz"
-
-    URL="https://github.com/golangci/golangci-lint/releases/download/${RELEASE}/golangci-lint-${RELEASE:1}-${os_type}-${ARCH}.tar.gz"
-    curl -sSL -o "${release_archive}" "${URL}"
-    echo ${sum} ${release_archive} | sha256sum --check --quiet -
-
-    release_tmp_dir="/tmp/${NAME}-${RELEASE}"
-    mkdir -p "${release_tmp_dir}"
-    tar -xzf "${release_archive}" --strip=1 -C "${release_tmp_dir}"
-
-    if [[ ! -f "${RELEASE_BINARY}" ]]; then
-      find "${BUILD_BIN}" -maxdepth 0 -regex '.*/'${NAME}'-[A-Za-z0-9\.]+$' -exec rm {} \;  # cleanup older versions
-      mv "${release_tmp_dir}/${NAME}" "${RELEASE_BINARY}"
-    fi
-
-    # Cleanup stale resources.
-    rm "${release_archive}"
-    rm -rf "${release_tmp_dir}"
+    env GOBIN="${BUILD_BIN}" go install github.com/golangci/golangci-lint/cmd/golangci-lint@"${RELEASE}"
```

Comment:
There's no ABI compatibility in Go, so there's no guarantee that a the golangci-lint binary will work with a different version of the go tooling. IIRC, that's what I saw when trying it, I can try to bring the binary back to see if things work.

With caching, the actual time difference is negligible, we can do this in a dedicated step to confirm.

Reply:
Ah ok I wasn't aware that golangci-lint relied on the go tooling, I thought it was a standalone binary. The cache itself in actions is just a tarball, I don't see how it could be any faster. I was mainly concerned for local dev.